### PR TITLE
apps/system/utils : Change to allocate array which saved terminated info

### PR DESCRIPTION
**Pull request by jeongchanKim**
terminated_infos can be affected with another configs, like CONFIG_MAX_TASKS.
So that it can also affect the stack size of stack monitor.
Changing that array to be allocated can avoid the problem.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [c04a52ffd0070f72753f09bfcf40fb7691c85ebd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c04a52ffd0070f72753f09bfcf40fb7691c85ebd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [614aca911ab317a35f8eaf32e23fdd93fff821d7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [614aca911ab317a35f8eaf32e23fdd93fff821d7] - Code Rule Check OK.

File: apps/system/utils/utils_stackmonitor.c [hunk 1]
```diff
@@ -132,15 +132,9 @@ static void stkmon_print_title(void)
...
-	struct stkmon_save_s terminated_infos[STKMON_MAX_LOGS];
-
-	/* Initialize the terminated_infos array. */
-	for (inactive_idx = 0; inactive_idx < (STKMON_MAX_LOGS); inactive_idx++) {	
-		terminated_infos[inactive_idx].timestamp = 0;
-	}
 
 	/* Copy the terminated task/pthread information from kernel. */
-	prctl((int)PR_GET_STKLOG, (struct stkmon_save_s *)terminated_infos);
+	prctl((int)PR_GET_STKLOG, terminated_infos);
```
Comment (Taejun-Kwon) [inline apps/system/utils/utils_stackmonitor.c]: prctl checks some of configs internally, and return error if each of config is not enabled.
Is it safe with out return value?

File: apps/system/utils/utils_stackmonitor.c [hunk 2]
```diff
@@ -109,7 +109,7 @@
  * Private Data
  ****************************************************************************/
 static volatile bool stkmon_started;
-
+static struct stkmon_save_s *terminated_infos;
```
Comment (Taejun-Kwon) [inline apps/system/utils/utils_stackmonitor.c]: You refined this as a global but using in only in the stkmon_print_inactive_list,
also you do not keep this as a static.
I can't understand why it refined as a global?

**Comment by seinfra** [comment]
Target : [78800d89611c7cb2d3c5b7aa78f2fbc0f9e7cda8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [78800d89611c7cb2d3c5b7aa78f2fbc0f9e7cda8] - Code Rule Check OK.

File: apps/system/utils/utils_stackmonitor.c [hunk 3]
```diff
@@ -109,7 +109,6 @@
  * Private Data
  ****************************************************************************/
 static volatile bool stkmon_started;
-
```
Comment (Taejun-Kwon) [inline apps/system/utils/utils_stackmonitor.c]: You'd better keep previous blank line here.

File: apps/system/utils/utils_stackmonitor.c [hunk 4]
```diff
@@ -205,6 +203,7 @@ static int stkmon_read_proc(FAR struct dirent *entryp, FAR void *arg)
 
 static void *stackmonitor_daemon(void *arg)
 {
+	struct stkmon_save_s *terminated_infos;
```
Comment (Taejun-Kwon) [inline apps/system/utils/utils_stackmonitor.c]: almost same opinion as my previous comment.
This function do not use 'terminated_infos', then it had better alloc & free in there.

File: apps/system/utils/utils_stackmonitor.c (hunk 3 above)
Comment (jeongchanKim) [reply apps/system/utils/utils_stackmonitor.c]: Okay. Thanks.

**Comment by seinfra** [comment]
Target : [9a78036e0151219ce3df27ca3ddc619401abd679] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9a78036e0151219ce3df27ca3ddc619401abd679] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [345a242caf7d4829d4e7a2a64169e77edb972fec] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [345a242caf7d4829d4e7a2a64169e77edb972fec] - Code Rule Check (C++) OK.

**Pull request merged**